### PR TITLE
fix: set usage features.amount minimum to 1

### PR DESCRIPTION
**Pull request by agreea**
## What Does this PR Do?
- fixes an issue with features where amounts could be < 1 but usage credit grants had to be amounts >= 1

---
## Summary by cubic
Enforces a minimum of 1 for UsageCreditGrant feature amounts across schema and runtime to prevent zero/negative grants. Replaces z.number().int() with core.safeZodPositiveInteger and adds focused tests.

- **Bug Fixes**
  - Added a runtime check in payment bookkeeping to throw when a usage credit grant amount is missing or <= 0.
  - Updated product lookup to return the product even if it has no prices instead of failing.

<sup>Written for commit 1dab6ad8c0edb4ebde4ecf0e4d78878c38426a7f. Summary will update automatically on new commits.</sup>

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved product pricing/feature retrieval with a safe fallback lookup to assemble results.
  * Enforced positive integer validation for usage credit grants and added runtime guard during payment processing.

* **Tests**
  * Added comprehensive feature schema tests covering insert/update, unions, and edge cases (amount, meter, frequency).
  * Strengthened payment-processing tests by directly seeding feature scenarios and added organization scoping in pagination tests.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: platform/flowglad-next/src/utils/bookkeeping/processPaymentIntentStatusUpdated.ts [hunk 1]
```diff
@@ -393,6 +393,14 @@ export const ledgerCommandForPaymentSucceeded = async (
     return undefined
   }
 
+  if (
```
Comment (joeysabs) [inline platform/flowglad-next/src/utils/bookkeeping/processPaymentIntentStatusUpdated.ts]: Should we also be verifying non-zero amounts for `grantEntitlementUsageCredits` as well?
